stats/statsreporter.go

```diff
@@ -88,24 +91,32 @@ func Post(destination string, stats []Sample) error {
 	payload := map[string][]Sample{"metrics": stats}
 	data, err := json.Marshal(payload)
 	if err != nil {
-		plog.Warningf("Couldn't marshal stats: ", err)
+		plog.WithField("stats", stats).WithError(err).
```

Comment:
The stats field is too verbose to log as is. Maybe log the size? e.g. "numstats": len(stats)